### DISCUSSION TITLE
Request to implement useChirality in RunReactants

### DISCUSSION THREAD

**Discussion by laurenreid1** [rdkit/rdkit#7292] [Development]
Hi RDKit developers,

Please can I request the implementation of the useChirality argument to the RunReactants function?

For info, I've attached a notebook demonstrating the issue (RDKit version 2023.09.6). When applying a chiral SMIRKS to a molecule, products are formed irrespective of the chirality in the reactants. It would be great if there was a useChirality argument that enforces chiral matching, similar to the functionality in HasSubstructMatch/ GetSubstructMatch (demonstrated in the second section of the notebook).

[RunReactants_example.zip](https://github.com/rdkit/rdkit/files/14743065/RunReactants_example.zip)

Thanks,

Lauren

**Comment by bp-kelley** [comment]
I agree this is a bit confusing, but until this is an option you can always filter your reactant lists as follows:

```
reactant = Chem.MolFromSmiles(...)
reactant.HasSubstructMatch(rxn.GetReactantTemplate(0), useChirality=True)
```

**Comment by laurenreid1** [comment]
Hi Brian,

Thanks for the suggestion. 

I considered this but realised if more than 1 match is present in the molecule, with one matching chirality and the other not, the reaction would still take place at both sites. I've managed to find a solution using a combination of GetSubstructMatches and setting protecting atoms on any atoms not included in a match. 

Thanks for the help.

Lauren
